src/eko/matching_conditions/n3lo/agg.py

```diff
@@ -0,0 +1,1556 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=too-many-lines
+import numba as nb
+import numpy as np
+
+from .aggTF2 import A_ggTF2_3
+
+
+@nb.njit("c16(c16,c16[:],c16[:],c16[:],c16[:],u4,f8)", cache=True)
+def A_gg_3(n, sx, smx, s3x, s4x, nf, L):  # pylint: disable=too-many-locals
+    r"""
+    Computes the |N3LO| singlet |OME| :math:`A_{gg}^{S,(3)}(N)`.
+    The experssion is presented in :cite:`Bierenbaum:2009mv`.
+
+    When using the code, please cite the complete list of references
+    available in :mod:`eko.matching_conditions.n3lo`.
+
+    Parameters
+    ----------
+        n : complex
+            Mellin moment
+        sx : numpy.ndarray
+            list S1 ... S5
+        smx : numpy.ndarray
+            list Sm1 ... Sm5
+        s3x : numpy.ndarray
+            list S21, S2m1, Sm21, Sm2m1
+        s4x : numpy.ndarray
+            list S31, S221, Sm22, Sm211, Sm31
+        nf : int
+            number of active flavor below the threshold
+        L : float
+            :math:`\ln(\mu_F^2 / m_h^2)`
+
+    Returns
+    -------
+        A_gg_3 : complex
+            :math:`A_{gg}^{S,(3)}(N)`
+
+    See Also
+    --------
+        A_ggTF2_3: eko.matching_conditions.n3lo.aggTF2.A_ggTF2_3
+            Incomplete part proportional to :math:`T_{F}^2`.
+    """
+    S1, S2, S3, S4 = sx[0], sx[1], sx[2], sx[3]
```

Comment:
now, I guess it is too late (though I'm not sure since this affects only the function head), but rather then this cryptic lines with random ordering, I rather would have made 3 arguments:
- `sx` a list with 11 elements: `s_{-5},...,s_{-1},nan,s_1,...s_5`
- `sxx` a 11x11 array: `s_{-5,-5},...` where most of the elements never get computed, but are simply nan or 0
- `sxxx` same in 3D

Reply:
If possible [`np.nan`](https://numpy.org/doc/stable/reference/constants.html?highlight=nan#numpy.nan), since it has a more explicit meaning.

Reply:
again, if you wish you can also separate by weight, but then please define a unique ordering somewhere, because by weight the sorting is not straight (you also can define an algorithm á là "biggest positive first + 1 number (31), next biggest + 1 number (22), then +2 number (211)" - but as said is a bit more trickier .. )

Reply:
And *if* you go for weights, please adjust the names: `s3x -> s_w3` (or similar)

Reply:
you are right, the order I had in mind was something like by weights but I too creative... So the more logical would be the one with arrays matrices and tensors, the more physical one the one with weights...